quickcomm/views.py

```diff
@@ -1,22 +1,33 @@
-from django.shortcuts import render, redirect
+from django.shortcuts import render, redirect, get_object_or_404
 from django.http import HttpResponse, HttpResponseRedirect
 from django.contrib.auth.decorators import login_required
 from django.contrib.auth import authenticate, login as auth_login, logout as auth_logout
-from quickcomm.forms import CreateMarkdownForm, CreatePlainTextForm, CreateLoginForm
+from django.core.paginator import Paginator
+from quickcomm.forms import CreateMarkdownForm, CreatePlainTextForm, CreateLoginForm, EditProfileForm
 from quickcomm.models import Author, Post
 
+
 # Create your views here.
 
+def get_current_author(request):
+    if request.user.is_authenticated:
+        author = Author.objects.get(user_id = request.user.id)
+    else:
+        author = None
+    return author
 
 def index(request):
+    current_author = get_current_author(request)
     context = {
-        'posts': Post.objects.all()
+        'posts': Post.objects.all(),
+        'current_author': current_author
     }
     return render(request, 'quickcomm/index.html', context)
 
 
 @login_required
 def create_post(request):
+    current_author = get_current_author(request)
```

Comment:
You should not need to send the current_author to the frontend every time. Since we are assuming that the person has to be logged in anyways, it is redundant.

For now, I think it can pass, and we can refactor later. It is a minor tweak and not that much of an issue.

Reply:
Are we assuming that they have to be logged in? I was under the impression that unregistered users could still browse authors and public posts. I may be incorrect though.